### PR TITLE
OHRM-725:Implement my-leave-entitlement API endpoint

### PR DESCRIPTION
**Pull request by RajithaKumara**

### CODE REVIEW

File: symfony/plugins/orangehrmRESTPlugin/lib/Api/Leave/MyLeaveRequestAPI.php [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+    }
+
+    /**
+     * @param $employeeId
+     * @return Response
+     * @throws InvalidParamException
+     * @throws RecordNotFoundException
+     * @throws \DaoException
+     */
+    public function getMyLeaveDetails($employeeId)
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmRESTPlugin/lib/Api/Leave/MyLeaveRequestAPI.php:143]: (int $employeeId): Response

File: symfony/plugins/orangehrmRESTPlugin/lib/Api/Leave/MyLeaveRequestAPI.php [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+                $leaveEntitlement = new LeaveEntitlement($entitlement->getId());
+                $leaveEntitlement->buildEntitlement($entitlement);
+                $leaveBalance = (array)$this->getLeaveEntitlementService()->getLeaveBalance($employeeId, $entitlement->getLeaveTypeId());
+                $leaveType = new LeaveType($entitlement->getLeaveTypeId(), $entitlement->getLeaveType()->getName());
+                array_walk($leaveBalance, function (&$value, $key) {
+                    $value = (float)$value;
+                });
+                $responseEntitlement[] = array_merge(
+                    $leaveEntitlement->toArray(),
+                    array(
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmRESTPlugin/lib/Api/Leave/MyLeaveRequestAPI.php:162]: Better to define the return structure array and filled the data

File: symfony/plugins/orangehrmRESTPlugin/lib/Api/Leave/MyLeaveRequestAPI.php [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace Orangehrm\Rest\Api\Leave;
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmRESTPlugin/lib/Api/Leave/MyLeaveRequestAPI.php:20]: Need to define api doc blocks as well

File: symfony/plugins/orangehrmRESTPlugin/test/orangehrmRESTPluginAllTests.php [hunk 4]
```diff
@@ -70,15 +70,16 @@ public static function suite() {
...
+        $suite->addTestFile(dirname(__FILE__) . '/api/model/ModelTraitTest.php');
 
-        return $suite;
-    }
+        // Api/Mobile
+        $suite->addTestFile(dirname(__FILE__) . '/api/mobile/ApiMyLeaveEntitlementAPITest.php');
+        $suite->addTestFile(dirname(__FILE__) . '/api/mobile/model/ApiLeaveEntitlementModelTest.php');
 
-    public static function main() {
-        PHPUnit_TextUI_TestRunner::run(self::suite());
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmRESTPlugin/test/orangehrmRESTPluginAllTests.php:80]: @samanthajayasinghe what is the requirement of this?

**Comment by RajithaKumara** [comment]
close because of open #661

**Pull request closed without merging**